### ISSUE TITLE
vc deconstruct error - more sample names in header than sample fields

### ISSUE THREAD

**Issue by jdamas13** [pangenome/pggb#287]
Hi,
I am trying to generate a pangenome using two genome assemblies of the same species.
I started by running partition_before_pggb, and now I am running the pggb command for each community.
My jobs are being killed at the vg deconstruct step. The message I am getting is shown below. 
I notice that for every chromosome, the run errors for the first line in the vcf file that has the CONFLICT flag.

Do you know why this is happening and how to fix it?
I appreciate any help you can provide.

```
[vg::deconstruct] decompose VCF
error: more sample names in header than sample fields
samples: query
line: ref#1#chr5  24977031        >61682>61685    A       C       60.0    .       AC=0;AF=0;AN=0;AT=>61682>61683>61685,>61682>61684>61685;CONFLICT=query;NS=0;LV=1;PS=>61254>62671   GT
Command exited with non-zero status 1
```

**Comment by AndreaGuarracino** [comment]
Hi @jdamas13, can you confirm you are using `vg` 1.40.0?

**Comment by ekg** [comment]
For clarity, this is not the most recent vg version. There has been a regression in vg deconstruct in recent versions, and only a specific range of versions, ending at 1.40.0, will work.

**Comment by jdamas13** [comment]
Hi, I was using the nf-core/pangenome dev docker container, which has vg: variation graph tool, version v1.40.0 "Suardi".

**Comment by vatanparast** [comment]
I got the same error using latest singularity version.

```[vg::deconstruct] making VCF with reference=Cantata and delim=# xxxxxxxxxxxxx Cantata:1000 ------------ 1000
vg deconstruct -P Cantata -H # -e -a -t 4 community.9/pg2-pg5_prefixed-50kb.community.9.fa.gz.bf3285f.11fba48.867196c.smooth.final.gfa
457.70s user 15.60s system 222% cpu 213.00s total 2875776Kb max memory
[vg::deconstruct] decompose VCF
vcfwave 1.0.7 processing...
error: more sample names in header than sample fields
samples: PG5
line: Cantata#1#Scf9YQZ_25_HRSCAF_39    19      >1823810>1823812        CC      C       60.0    .       AC=0;AF=0;AN=0;AT=<1823812<1823811<1823810,<1823812<1823810;NS=0;LV=0   GT
Command exited with non-zero status 1
```
vg:
  version:            v1.40.0
  deconstruct:        Cantata:1000
reporting:
  version:            v1.21
  multiqc:            true

**Comment by glennhickey** [comment]
This looks to be the same error as https://github.com/ComparativeGenomicsToolkit/cactus/issues/1416 and possibly https://github.com/ComparativeGenomicsToolkit/cactus/issues/1402

I'm looking into it now, and it seems to be caused by:
* `vg deconstruct` writes genotype as `.`
* then `vcfbub` replaces the `.` genotype with a completely empty column, producing an invalid VCF.

It seems strange that this error is only now coming up, as `vg deconstruct` and `vcfbub` haven't changed much at all lately (tho `deconstruct` will be very refactored in the next vg release). 

I am going to double-check the `deconstruct` end today (I think the . genotype is coming from its conflict resolution and is probably by design).  But it seems like there is a bug in `vcfbub` (by way of the api its using to write VCF) that by stripping `.` genotype columns produces invalid VCF.  @ekg @AndreaGuarracino let me know if you want some data to reproduce.

**Comment by ekg** [comment]
Would love data please send!

**Comment by AndreaGuarracino** [comment]
This was nicely fixed by @glennhickey in `vcfbub` v0.1.1!
